File: coding/src/com/algos/li25_greedy/L605_CanPlaceFlowers.java
package edu.algos.li25_greedy;

public class L605_CanPlaceFlowers {

	public static void main(String[] args) {
		L605_CanPlaceFlowers obj = new L605_CanPlaceFlowers();

		int[] flowerbed = new int[] {1,0,0,0,0,1};
		System.out.println(obj.canPlaceFlowers(flowerbed, 2));
	}

	// Problem: https://leetcode.com/problems/can-place-flowers/
	// Idea: Just skip the cell if i, i-1, i+1 positions contains 1
	public boolean canPlaceFlowers(int[] flowerbed, int n) {
		for (int i = 0; i < flowerbed.length; i++) {
			if (flowerbed[i] == 1) continue;
			if (i>0 && flowerbed[i - 1] == 1)  continue;
            if (i<flowerbed.length-1 && flowerbed[i + 1] == 1)  continue;

			flowerbed[i] = 1;
			n--;
		}

		return n <= 0;
	}

}

File: dsa/concepts/21_greedy/python/l605_can_place_flowers.py
# Problem: https://leetcode.com/problems/can-place-flowers/
# Idea: Just skip the cell if i, i-1, i+1 positions contains 1

def canPlaceFlowers(flowerbed, n):
    length = len(flowerbed)

    for i in range(length):
        if flowerbed[i] == 1:
            continue
        if i > 0 and flowerbed[i - 1] == 1:
            continue
        if i < length - 1 and flowerbed[i + 1] == 1:
            continue

        flowerbed[i] = 1
        n -= 1

    return n <= 0


# -------- Test --------
print(canPlaceFlowers([1, 0, 0, 0, 0, 1], 2))  # Expected: False
